xstructs/map.go

```diff
@@ -11,20 +11,78 @@ import (
 // tagCategories is a list of tag categories to check for.
 var tagCategories = []string{"json", "yaml"}
 
+// WithTags allows you to specify custom tag categories to check for.
+// It can be used to override the default "json" and "yaml" tags.
+// The tags are checked in the order they are provided.
+func WithTags(tags ...string) ToMapOptions {
+	return func(h *handler) {
+		h.tags = tags
+	}
+}
+
+// WithAllowNoTags allows you to specify whether to allow fields without tags.
+// If used, fields without tags will be included in the output map.
+func WithAllowNoTags() ToMapOptions {
+	return func(h *handler) {
+		h.allowNoTags = true
+	}
+}
+
+// ToMapOptions is a function that modifies the handler.
+type ToMapOptions func(*handler)
```

Comment:
I think the norm is singular (`ToMapOption`). It is throughout stdlib this far, anyway. So maybe for consistency, make it singular.